tests/unit-mpi/test_base.py

```diff
@@ -0,0 +1,108 @@
+    elif rank == 1:
+        data = comm.recv(source=0, tag=0)
+        assert data["msg"] == "Hello from rank 0"
+
+
+@pytest.mark.parametrize("create_tmp_simulation_config_file", [
+    {
+        "simconfig_fixture": "ringtest_baseconfig",
+        "extra_config": {
+            "target_simulator": "NEURON",
+            "inputs": {
+                "Stimulus": {
+                    "module": "pulse",
+                    "input_type": "current_clamp",
+                    "delay": 5,
+                    "duration": 50,
+                    "node_set": "RingA",
+                    "represents_physical_electrode": True,
+                    "amp_start": 10,
+                    "width": 1,
+                    "frequency": 50
+                }
+            },
+            "reports": {
+                "soma_v": {
+                    "type": "compartment",
+                    "cells": "Mosaic",
+                    "variable_name": "v",
+                    "sections": "soma",
+                    "dt": 0.1,
+                    "start_time": 0.0,
+                    "end_time": 18.0,
+                },
+                "compartment_i": {
+                    "type": "compartment",
+                    "cells": "Mosaic",
+                    "variable_name": "i_membrane",
+                    "sections": "all",
+                    "dt": 1,
+                    "start_time": 0.0,
+                    "end_time": 40.0,
+                },
+            },
+        }
+    },
+    {
+        "simconfig_fixture": "ringtest_baseconfig",
+        "extra_config": {
+            "target_simulator": "CORENEURON",
+            "inputs": {
+                "Stimulus": {
+                    "module": "pulse",
+                    "input_type": "current_clamp",
+                    "delay": 5,
+                    "duration": 50,
+                    "node_set": "RingA",
+                    "represents_physical_electrode": True,
+                    "amp_start": 10,
+                    "width": 1,
+                    "frequency": 50
+                }
+            },
+            "reports": {
+                "soma_v": {
+                    "type": "compartment",
+                    "cells": "Mosaic",
+                    "variable_name": "v",
+                    "sections": "soma",
+                    "dt": 0.1,
+                    "start_time": 0.0,
+                    "end_time": 18.0,
+                },
+                "compartment_i": {
+                    "type": "compartment",
+                    "cells": "Mosaic",
+                    "variable_name": "i_membrane",
+                    "sections": "all",
+                    "dt": 1,
+                    "start_time": 0.0,
+                    "end_time": 40.0,
+                },
+            },
+        }
+    }
+], indirect=True)
+@pytest.mark.mpi(ranks=[1, 2, 4])
+def test_neurodamus(create_tmp_simulation_config_file, mpi_ranks, mpi_tmp_path):
```

Comment:
We can add checks here:
1. assert the local gids spread over all ranks, e.g.
```
node = n.circuits.get_node_manager("RingA")
local_gids = node.local_nodes.final_gids()
npt.assert_allclose(local_gids, ref_gids) 
```
`ref_gids` are different in each rank. 
2. check the MPI size
```
from neurodamus.core import MPI
assert MPI.size == mpi_ranks
```
3. with ranks = [4] for the coreneuron simulation,  a fake cell is expected to be added in rank id 3, so if we run  ```n = Neurodamus(create_tmp_simulation_config_file, keep_build=True)```, we should see `2004_1/2/3).dat` in the `coreneuron_input` folder.

Reply:
And what is `mpi_tmp_path` ? I see the `pytest-isolate-mpi` docs says it wraps the pytest `tmp_path` , are they the same value? So our `SimConig.output_root` `is_relative_to(mpi_tmp_path)` ?

Reply:
they are not the same value since it fails if I sue the old one. I think `pytest-isolate-mpi` adds a folder on the way for collecting the same mpi run